apk-generator/index.html

```diff
@@ -121,13 +121,19 @@ <h3 id="status"></h3>
                 $('#generator-progress').css('display', 'block')
                 $('#upload-progress').css('display', 'block')
                 var ary = $(form).serializeArray();
+                var emailse = $("#Email").val();
                 var obj = {};
                 updatePercentUpload(10);
                 for (var a = 0; a < ary.length; a++) obj[ary[a].name] = ary[a].value;
                 console.log("JSON", obj);
                 if (obj.Email == "" || obj.App_Name == "") {
```

Comment:
This will in fact make a good user experience.
Saying user enters wrong email and empty API link, so according to your
implementation he will be shown 2 alerts,which is not a good ux.

Speaking of your point, sure we can remove the line where whole page
refreshes on wrong input.
```setTimeout("location.reload(true);", 1);```
That should address the issue of user having to fill the whole form again.

>

-- 
Best,
Harshit Dwivedi
Contact : +919968195588
GitHub <https://github.com/the-dagger> | LinkedIn
<https://in.linkedin.com/in/hadwivedi>
Google Play <https://goo.gl/wl3loD> | Facebook
<https://www.facebook.com/daggerdwivedi>